modules/ui/sidebar.js

```diff
@@ -412,6 +418,40 @@ export function uiSidebar(context) {
                 });
         };
 
+        /*
+            The checkWindowWidth function checks the width of the working window when the mouse is over the
+            working area and flagCollapse and flagExpand is to prevent the re-triggering action after the window
+            is resized if the window is resized and the flags are not used then whenever we will try to expand
+            the side bar when the window width is below 700 then it will automatically collapse the side bar
+            which is not what we want.
+        */ 
+            function checkWindowWidth()
+            {
+                containerWidth = container.node().getBoundingClientRect().width;
+    
+                /*
+                    this is to reset the flag every time we resize the window
+                    so that the functionality offered by the function is not just for the first time the
+                    iD is loaded.
```

Comment:
```suggestion
                    Reset `flagCollapse` every time we resize the window
                    otherwise the behavior would only apply the first time iD is loaded.
```